docs/src/main/asciidoc/picocli.adoc

```diff
@@ -0,0 +1,178 @@
+////
+This guide is maintained in the main Quarkus repository
+and pull requests should be submitted there:
+https://github.com/quarkusio/quarkus/tree/master/docs/src/main/asciidoc
+////
+= Quarkus - Command Mode with Picocli
+:extension-status: experimental
+
+include::./attributes.adoc[]
+
+https://picocli.info/[Picocli] is an open source tool for creating rich command line applications.
+
+Quarkus provides support for using Picocli. This guide contains examples of `picocli` extension usage.
+
+include::./status-include.adoc[]
+
+IMPORTANT: If you are not familiar with the Quarkus Command Mode, consider reading the link:command-mode-reference[Command Mode Applications] first.
+
+== Configuration
+
+Once you have your Quarkus project configured you can add the `picocli` extension
+to your project by running the following command in your project base directory.
+
+[source,bash]
+----
+./mvnw quarkus:add-extension -Dextensions="picocli"
+----
+
+This will add the following to your pom.xml:
+
+[source,xml]
+----
+    <dependency>
+        <groupId>io.quarkus</groupId>
+        <artifactId>quarkus-picocli</artifactId>
+    </dependency>
+----
+
+== Simple Picocli Application with one Command
+
+Simple PicocliApplication with only one Command can be created as follows:
```

Comment:
```suggestion
Simple Picocli Application with only one Command can be created as follows:
```

Reply:
not really fan of calling this "Picocli Application" it is still an Quarkus Application just enhanced with command line parsing done by picocli.

Reply:
How about something like this:

> == Standalone picocli-based command line application
> A standalone command line application with a single picocli command can be created as follows:

I'm finding the term "standalone" command/application, useful when distinguishing from a "command suite" with subcommands.

Reply:
> not really fan of calling this "Picocli Application" it is still an Quarkus Application just enhanced with command line parsing done by picocli.

+1



> How about something like this:
> 
> > == Standalone picocli-based command line application
> > A standalone command line application with a single picocli command can be created as follows:
> 
> I'm finding the term "standalone" command/application, useful when distinguishing from a "command suite" with subcommands.

But can't we also create subcommands with this extension? I think calling simply a `command line application` may be good enough

Reply:
> But can't we also create subcommands with this extension?

I was thinking to have a separate section in the documentation for a command with subcommands; both are common scenarios and it is helpful to have clear examples for both.